src/Servers/Kestrel/Core/src/Adapter/Internal/LoggingStream.cs

```diff
@@ -160,7 +160,14 @@ private void Log(string method, ReadOnlySpan<byte> buffer)
             // Write the bytes as if they were ASCII
             for (int i = 0; i < buffer.Length; i++)
             {
-                rawDataBuilder.Append((char)buffer[i]);
+                var bufferChar = (char)buffer[i];
```

Comment:
This caused a failure in xunit test reporting due to control characters appearing in logs, so I escaped the non-printable characters @halter73 Any objections to this change?

Reply:
This looks good. Can you include a sample of the output?

Prior improvements here: https://github.com/aspnet/KestrelHttpServer/pull/3046/files

Reply:
```
dbug: Microsoft.AspNetCore.Server.Kestrel.Core.Adapter.Internal.LoggingConnectionAdapter[0]
      WriteAsync[129] 00 00 61 01 04 00 00 00 03 88 00 04 64 61 74 65 1D 4D 6F 6E 2C 20 31 37 20 4A 75 6E 20 32 30 31 39 20 32 32 3A 31 31 3A 32 31 20 47 4D 54 00 0C 63 6F 6E 74 65 6E 74 2D 74 79 70 65 0A 74 65 78 74 2F 70 6C 61 69 6E 00 06 73 65 72 76 65 72 07 4B 65 73 74 72 65 6C 00 0E 63 6F 6E 74 65 6E 74 2D 6C 65 6E 67 74 68 02 31 34 00 00 0E 00 00 00 00 00 03 68 65 6C 6C 6F 2C 20 77 6F 72 6C 64 0D 0A
      \x00\x00a\x01\x04\x00\x00\x00\x03\x88\x00\x04date\x1DMon, 17 Jun 2019 22:11:21 GMT\x00\x0Ccontent-type\x0Atext/plain\x00\x06server\x07Kestrel\x00\x0Econtent-length\x0214\x00\x00\x0E\x00\x00\x00\x00\x00\x03hello, world\x0D\x0A
```

Reply:
That works. Was that an HTTP/2 response?